### DISCUSSION TITLE
Gatsby schema customisation issue with Contentful reference types

### DISCUSSION THREAD

**Discussion by Tmcerlean** [gatsbyjs/gatsby#35895] [Help]
**Issue**

- Attempting to enable Contentful blog posts to be authored by "internal person" and "external person" content models.
- Currently they are authored by a regular "person" content model.
- I have created dummy blog post content (by both an internal and external person) to manually try and populate the union graphQL fields. I think this may have only populated "ContentfulInternalPerson" and "ContentfulExternalPerson" types individually and not the union.
- I am running into the following error:

> There was an error in your GraphQL query:
> 
> Cannot query field "name" on type "ContentfulInternalPersonContentfulExternalPersonUnion".
> 
> If you don't expect "name" to exist on the type "ContentfulInternalPersonContentfulExternalPersonUnion" it is most likely a typo.
> However, if you expect "name" to exist there are a couple of solutions to common problems:
> 
> If you added a new data source and/or changed something inside gatsby-node.js/gatsby-config.js, please try a restart of your development server
> The field might be accessible in another subfield, please try your query in GraphiQL and use the GraphiQL explorer to see which fields you can query and what shape they have
> You want to optionally use your field "name" and right now it is not used anywhere. Therefore Gatsby can't infer the type and add it to the GraphQL schema. A quick fix is to add at least one entry with that field ("dummy content")
> It is recommended to explicitly type your GraphQL schema if you want to use optional fields. This way you don't have to add the mentioned "dummy content". Visit our docs to learn how you can define the schema for "ContentfulInternalPersonContentfulExternalPersonUnion":
> https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization#creating-type-definitions
> 
> For more details see https://gatsby.dev/issue-how-to

- I can see that Gatsby is automatically creating a Union type of the two types from Contentful. But as the Union isn't populated, Gatsby hasn't dynamically built any schema for it and so it can't be queried.

**What I have tried**

- I have tried to manually populate dummy data for the Union type.
- I have also tried to define the schema manually following the link in the docs.
- When I did this, I ran into the following error:

> Missing onError handler for invocation 'building-schema', error was 'Error: Schema must contain uniquely named     
> types but contains multiple types named "ContentfulInternalPersonContentfulExternalPersonUnion".'. Stacktrace was 'Error:  
> Schema must contain uniquely named types but contains multiple types named
> "ContentfulInternalPersonContentfulExternalPersonUnion".

- I can see that I've duplicated this union type and therefore prevented Gatsby from building, but I'm unclear on how this can be avoided.

**Comment by LarryKarani** [comment]
were you able to solve this? getting these same error
